### ISSUE TITLE
Documentation wanted

### ISSUE THREAD

**Issue by SQReder** [iamtabrezkhan/awesome-react-generator#24]
1. Is it extendable? 
    1. Can I add extra templates or override current? 
    1. Which template engine is used?
    1. Can I make templates for style/test file?
1. Can I override some of CLI params? For example, I want test files to be generated each time

**Comment by iamtabrezkhan** [comment]
Hi @SQReder,

It is not extendable as of now and we are not using any template engine instead we are using plain boilerplate files (copy pasting them).
No, you cannot override CLI params. You will have to give `--test` option every time you want to generate test file.

Great questions! We could use some config file or package.json to override some CLI behaviour.
I would be more than happy to know your thoughts.

**Comment by SQReder** [comment]
Did you made a research over existing file generation solutions? Tons of them 🙂

I currently use integrated file templates feature in WebStorm, which uses Apache Template as a template language. The problem is that these templates are not shareable with the team, who's can use VSCode. This can be solved with a configurable component generator.

Telling about configuration I think it's good to have a dedicated top-level directory for all the configuration stuff as, for example, Storybook does.

The configuration may be like
```typescript
{
    "tests": {
        "template": "mocha" | "jest" | string,
        "alwaysGenerate": true,
    },
    "style": {
        "template": "css" | "less" | "scss" | string,
        "alwaysGenerate": false,
    }
}
```

Custom templates can be stored in a subdirectory of configuration directory and definitely must use some templating engine with a bunch of variables defined like "FileName", "Path" and so on.

**Comment by SQReder** [comment]
I think you can look for inspiration in [generate-react-cli](https://github.com/arminbro/generate-react-cli) project

**Comment by iamtabrezkhan** [comment]
Yeah, makes sense. Using a templating engine will give more control to the users of the CLI.
I am planning to look into ejs and see how it goes from there.
Really appreciate your input :+1:
